airmozilla/main/static/main/js/calendar.js

```diff
@@ -1,31 +1,34 @@
 $(function() {
-    var weekends, firstDay = 0;
-    var x = new Date();
-    var currentTimeZoneOffsetInHours = x.getTimezoneOffset() / 60;
-    if (currentTimeZoneOffsetInHours <= -7) {
-        // sufficiently far east that some Friday events might actually
-        // start on a Saturday
-        weekends = true;
-        firstDay = 1;  // far-east people prefer Monday
-    } else {
-        weekends = false;
-    }
+    var weekends = true;
     var container = $('#calendar');
-    container.fullCalendar({
-        events: {
-            url: container.data('events-url'),
-            cache: true
-        },
-        timezone: 'local',
-        weekends: weekends,
-        firstDay: firstDay,
-        allDaySlot: false,
-        defaultTimedEventDuration: '01:00:00',
-        timeFormat: 'h:mm a',
-        header: {
-            left: 'prev,next today',
-            center: 'title',
-            right: 'month,agendaWeek,agendaDay'
-        }
+    var startsOnMonday = $('input[type="checkbox"]');
+    // if user isn't in American TZ then Monday is first day of week
+    var tz = jstz.determine().name();
+    if (!tz.match('^America/')) {
+        startsOnMonday.prop('checked', true);
+    }
+    initCalendar = function() {
+        container.fullCalendar({
+            events: {
+                url: container.data('events-url'),
+                cache: true
+            },
+            timezone: 'local',
+            weekends: weekends,
+            firstDay: startsOnMonday.prop('checked')? 1 : 0,
+            allDaySlot: false,
+            defaultTimedEventDuration: '01:00:00',
+            timeFormat: 'h:mm a',
+            header: {
+                left: 'prev,next today',
+                center: 'title',
+                right: 'month,agendaWeek,agendaDay'
+            }
+        });
+    };
+    startsOnMonday.on('change', function(){
+        container.fullCalendar('destroy');
+        initCalendar();
```

Comment:
Instead of re-creating the whole thing, can you see if it can be done with http://fullcalendar.io/docs/display/render/